src/panels/lovelace/cards/hui-entity-button-card.ts

```diff
@@ -115,6 +115,9 @@ class HuiEntityButtonCard extends LitElement implements LovelaceCard {
           style="${styleMap({
             filter: this._computeBrightness(stateObj),
             color: this._computeColor(stateObj),
+            height: this._config.icon_height
```

Comment:
The config should contain a valid CSS value, so it should be stored with `px` already added to it. The UI editor should only support a value that ends in `px` and should allow editing it as a number, not exposing the `px` to the user. That way, advanced users can still use a `%` or other CSS values in their YAML.

Reply:
I'm not liking this idea. So I have to use the YAML editor in order to use `%`? And what happens if I use `%` via the raw config editor and then open the UI editor?

I don't think we should mess with the user's input. Just use what they give us. Perhaps we can do better validation in the future, but I don't think it's necessary.

Reply:
So I'll add a suffix to the UI side and allow the yaml to edited as the advanced user likes then just pull in the setting as is on the card.

Reply:
OK I seem to have it working.

To go on @iantrich 's point. I've managed to get it so it leaves this field alone but without messing with the 'advanced' input when you switch back to or save and reopen the config UI.